clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -2064,6 +2064,9 @@ llvm::Value *CodeGenFunction::EmitToMemory(llvm::Value *Value, QualType Ty) {
 
   if (Ty->isExtVectorBoolType()) {
```

Comment:
you replaced a bunch of `isExtVectorBoolType()` with `isPackedVectorBoolType`.  We are only doing HLSL modifications on the `isExtVectorBoolType()`. But `isExtVectorBoolType` doesn't mean the vector is not not packed. Is the Zero extend and truncation to get them into a form that they will become packed?